### PR TITLE
Add Cleanup Stale Branch Action

### PR DESCRIPTION
**Pull request by 4lch4**
This PR adds a workflow for the [Cleanup Stale Branches Action][0] we created for Project O11y.

It will run every 6 hours and determine if any stale branches were detected and need to be flagged for deletion.

[0]: https://github.com/liatrio/cleanup-stale-branches-action

### CODE REVIEW

File: .github/workflows/stale_branches.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: Run O11y Stale Branch POC
+
+on:
+  workflow_dispatch: {}
+  schedule:
+    - cron: '0 */6 * * *'
```
Comment (adrielp) [inline .github/workflows/stale_branches.yml:6]: Let's update this to once a day, before or after working hours.

File: .github/workflows/stale_branches.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+name: Run O11y Stale Branch POC
+
+on:
+  workflow_dispatch: {}
```
Comment (adrielp) [inline .github/workflows/stale_branches.yml:4]: Let's remove this and just let it be automatic. If we find we want to be able to trigger manually later on, we can add this back.

File: .github/workflows/stale_branches.yml (hunk 1 above)
Comment (4lch4) [reply .github/workflows/stale_branches.yml]: This has been done so I'll resolve this conversation.

File: .github/workflows/stale_branches.yml (hunk 2 above)
Comment (4lch4) [reply .github/workflows/stale_branches.yml]: This has been done so I'll resolve this conversation.

**Pull request closed without merging**